game.py

```diff
@@ -1,41 +1,84 @@
-from random import randint
+from random import choice, randint
 
 from exceptions import NimException
 
 
 class Bot():
+    bot_type_list = []
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        cls.bot_type_list.append(cls)
+
+    def valid_pile_indexes(self, state):
+        return [i for i, pile in enumerate(state) if pile >= 1]
+
     def move(self, state):
-        raise NotImplementedError
+        if not self.valid_pile_indexes(state):
+            raise NimException(f'No valid moves on state: {state}')
+        return self.move_strategy(state)
 
 
 class Random(Bot):
     """
-    Cory's bot :)
+    Bot that always chooses a random number of stones from a random pile
     """
-    pass
+    def move_strategy(self, state):
+        pile_index = choice(self.valid_pile_indexes(state))
+        stones = randint(1, state[pile_index])
+        return {'pile': pile_index, 'stones': stones}
 
 
 class Perfect(Bot):
     """
     Hunter's bot :)
+    Temporary strategy until implemented
     """
-    pass
+    def move_strategy(self, state):
+        for i, pile in enumerate(state):
+            if pile >= 1:
+                return {'pile': i, 'stones': 1}
```

Comment:
You should rewrite all of those similar methods to use the `valid_pile_indexes` method.
I know it's just a placeholder, but might as well.